frontend/src/tests/setup.ts

```
@@ -161,3 +161,18 @@ expect.extend(matchers as any);
 afterEach(() => {
   cleanup();
 });
+
+window.HTMLElement.prototype.scrollIntoView = vi.fn();
+
+// Mock IntersectionObserver
+class IntersectionObserver {
+  // biome-ignore lint/suspicious/noExplicitAny: test double
+  constructor(_cb: any = vi.fn(), _options: any = {}) { }
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)